algorithms/unit_tests/TestStdAlgorithmsCommon.hpp

```diff
@@ -254,11 +254,12 @@ auto make_bounds(const ValueType1& lower, const ValueType2 upper) {
   return Kokkos::pair<ValueType1, ValueType2>{lower, upper};
 }
 
-#if defined(__GNUC__) && __GNUC__ == 8
+#if defined(_GLIBCXX_RELEASE) && (_GLIBCXX_RELEASE <= 9)
 
-// GCC 8 doesn't come with reduce, transform_reduce, exclusive_scan,
-// inclusive_scan, transform_exclusive_scan and transform_inclusive_scan so here
-// are simplified versions of them, only for testing purpose
+// GCC 8 does not have reduce, transform_reduce, exclusive_scan, inclusive_scan,
```

Comment:
```suggestion
// libstdc++ as provided by GCC 8 does not have reduce, transform_reduce, exclusive_scan, inclusive_scan,
```

I think that is the specification that the code actually needs to make. The standard library implementation and compiler are separate, even if they're shipped and versioned together. Clang definitely can compile libstdc++, and I expect GCC can compile LLVM/Clang's libc++.

Reply:
Thanks @PhilMiller i agree, I will make the changes !